### PR TITLE
[MPDX-8340] Fix appeal mass selection logic

### PR DESCRIPTION
**Pull request by canac**
## Description

There were various mass-selection bugs, especially in appeals flows view. This PR makes the mass selection logic much more robust and ensures that the total count and checkbox state is perfectly in sync with the selected ids.

[MPDX-8340](https://jira.cru.org/browse/MPDX-8340)

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx [hunk 1]
```diff
@@ -178,9 +178,25 @@ export const AppealsProvider: React.FC<AppealsContextProps> = ({
     },
     skip: contactCount === 0,
   });
+  // In the flows view, we need the total count of all contacts in every column, but the API
+  // filters out contacts excluded from an appeal. We have to load excluded contacts also and
+  // manually merge them with the other contacts.
+  const { data: allExcludedContacts } = useGetIdsForMassSelectionQuery({
```
Comment (dr-bizz) [inline src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx:184]: We grab the excluded contacts on line `275`, maybe you could use that query instead?

**Review by dr-bizz — COMMENTED** [review]
Looking really good. I left a comment, but it's working well.

File: src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx (hunk 1 above)
Comment (canac) [reply src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx:184]: That query just loads the total contact count. We need the contact ids for mass selection.

**Review by dr-bizz — APPROVED** [review]

**Comment by dr-bizz** [comment]
Great work!

**Pull request merged**